### DISCUSSION TITLE
How to create a thread channel?

### DISCUSSION THREAD

**Discussion by curswqq** [discord-jda/JDA#2632] [Questions and Help]
Question: how to create a thread in a text channel and interact with it (add users, etc.). I couldn't figure it out myself, so I'm waiting for your answer.

**Comment by curswqq** [comment]
wating for one week answer, cringe

**Comment by bruskylikescoding** [comment]
Hey! Here is a very easy solution for your problem. After a command has been issued, the bot will send a text message to the channel and create a thread onto this. You can of course choose other messages from members, but you will need a message to start a thread channel.

```java
@Override
    public void onSlashCommandInteraction(SlashCommandInteractionEvent event) {
        if(event.getName().equals("test")) {
              Message message = event.getMessageChannel().sendMessage("Creating new Thread!").complete();
              ThreadChannel threadChannel = message.createThreadChannel("Cool thread channel!").complete();
  
              threadChannel.addThreadMember(event.getMember()).queue();
        }
    }
```

![image](https://github.com/discord-jda/JDA/assets/97536100/8e169550-e88a-42bc-8532-6dad8243bb8b)
![image](https://github.com/discord-jda/JDA/assets/97536100/59084795-ad7c-4ac2-a568-3a38414b32e4)
